### PR TITLE
added support for decision_function, that was needed for example in a cross validation with auc_roc scoring

### PR DESCRIPTION
**Pull request by zbenmo**
Re: Issue #511

### CODE REVIEW

**Comment by koaning** [comment]
Could you also add a unit test that confirms https://github.com/koaning/scikit-lego/issues/511 won't re-occur?

File: tests/test_meta/test_grouped_predictor.py [hunk 1]
```diff
@@ -557,3 +557,11 @@ def test_missing_check():
     with pytest.raises(ValueError) as e:
         GroupedPredictor(model, groups = ['diet']).fit(X, y)
         assert "contains NaN" in str(e)
+
+
+def test_has_decision_function():
+    df = load_chicken(as_frame=True)
+
+    X, y = df.drop(columns='weight'),  df['weight']
+    # This should NOT raise errors
```
Comment (koaning) [inline tests/test_meta/test_grouped_predictor.py:566]: Could you add a comment that links to the original issue?

**Comment by koaning** [comment]
The implementation looks good to me but @MBrouns figured I'd loop you in as well.

**Comment by koaning** [comment]
Some of the unit tests here seem to be breaking that are unrelated to this PR. I'm working on a fix here: https://github.com/koaning/scikit-lego/pull/512.

**Comment by koaning** [comment]
Just fixed the main branch.

**Comment by MBrouns** [comment]
I'm okay with the implementation for now, but we should probably remove the duplication between `decision-function`, `predict` and `predict_proba` at some point. Doesn't have to be in this PR though

**Comment by koaning** [comment]
Works for me 👍 

Merged it and I'm preparing a new patch release now.

**Comment by zbenmo** [comment]
Thanks. Was a fun experience to make a small contribution to scikit-lego.

**Pull request merged**